DataCleaningScripts/new_moon_numbers.r

```diff
@@ -32,54 +31,66 @@ find_first_trap_night = function(dat) {
 #' @param target_date single date object
 #' @param newmoondates vector of date objects
 #' 
-#' @example closest_newmoon(as.Date('2017-02-26'),moon_dates$NewMOonDate)
+#' @example closest_newmoon(as.Date('2017-02-26'),moon_dates$NewMoonDate)
 #' 
 closest_newmoon = function(target_date,newmoondates) {
   closest = which.min(abs(target_date-newmoondates))
   return(closest)
 }
 
-#' Updates existing version of moon_dates.csv with latest census numbers, rather than recreate the whole file
+#' Updates existing version of moon_dates.csv with latest census numbers
 #'
 #'
 #'
-#' @param final_data_location file path where csv of final data should be saved (in the PortalData repo)
+#' @param
 #' 
-#' @example update_moon_dates(final_data_location='C:/Users/EC/Desktop/git/PortalData/Rodents/moon_dates.csv')
+#' @example update_moon_dates()
 
-update_moon_dates = function(final_data_location) {
+update_moon_dates = function() {
   # load existing moon_dates.csv file
-  moon_dates=read.csv(text=getURL("https://raw.githubusercontent.com/weecology/PortalData/master/Rodents/moon_dates.csv"),stringsAsFactors = F)
+  moon_dates=read.csv("../Rodents/moon_dates.csv",stringsAsFactors = F)
+  moon_dates$NewMoonDate = as.Date(moon_dates$NewMoonDate)
   moon_dates$CensusDate = as.Date(moon_dates$CensusDate)
   # load rodent trapping data
-  trappingdat=read.csv(text=getURL("https://raw.githubusercontent.com/weecology/PortalData/master/Rodents/Portal_rodent_trapping.csv"))  
+  trappingdat=read.csv("../Rodents/Portal_rodent_trapping.csv")  
   trappingdat$CensusDate = as.Date(paste(trappingdat$Year,trappingdat$Month,trappingdat$Day,sep='-'))
+  # load new moon dates
+  newmoons=read.csv(url('https://www.dropbox.com/s/z9kq7qvu2r7uwzu/moon-phases-1977-2018-America_Phoenix.csv?dl=1'),header = T, stringsAsFactors = F) %>%
```

Comment:
Looks like we can acquire the new moon data programmatically from here:
http://aa.usno.navy.mil/data/docs/MoonPhase.php

Using something based on:

```
library(htmltab)
data = htmltab(doc="http://aa.usno.navy.mil/cgi-bin/aa_phases.pl?year=2017&month=6&day=30&nump=50&format=t")```
data$"Date and Time (Universal Time)"[data$"Moon Phase" == "New Moon"]
```

The year, month, day, and nump (number of values), values in the url can be set to give us the general set of dates we need.